Adding the tests

kristinapeneva opened the pull request:
Hello, guys. My tests are ready.
There are 2 issues:
1. The tests pass one by one, but 2nd and 3rd test fail, when run together. (org.openqa.selenium.ElementClickInterceptedException: element click intercepted)
2. I cannot run the test on my laptop without adding WebDriverManager.chromedriver().browserVersion("97").setup();

Please, run the tests on your laptop and let me know how it goes.

kristinapeneva committed:
Adding the tests

gercen8905 reviewed (commented):
Awesome tests!
Some comments:
1. Test in @Test and in the name of test.
2. You can add the third parameter inside assertion, message if the test fails.
3. xpath in line 93 could be shorter (you may use contains or starts with).
4. DRY: you use deliverToButton and clicking it three times. It could be the small method. And it's not very big upgrade). Just for "readability"
5. Raw use of List -> List<WebElements>
6. Deprecated constructor in 45, 60, 78. Duration class instead

czajkaaga commented on src/test/java/com/epam/qalab/DeliverToTest.java:
```diff
@@ -0,0 +1,101 @@
+        WebDriverWait wait = new WebDriverWait(driver, 10);
+        wait.until(ExpectedConditions.elementToBeClickable(By.xpath("//span[@class='a-dropdown-container']")));
+        Select select = new Select(driver.findElement(By.xpath("//select[@class='a-native-dropdown']")));
+        select.selectByVisibleText("Bulgaria");
```
you can put "Bulgaria" as the variable and use that variable in the assertion

czajkaaga commented on src/test/java/com/epam/qalab/DeliverToTest.java:
```diff
@@ -0,0 +1,101 @@
+        WebElement deliverToButton = driver.findElement(By.id("nav-global-location-popover-link"));
+        deliverToButton.click();
+
+        WebElement zipCodeField = new WebDriverWait(driver, Duration.ofSeconds(10)).until(ExpectedConditions.visibilityOfElementLocated(By.id("GLUXZipUpdateInput")));
```
small note, this line can be split into two lines, then we don't need to scroll horizontally to see what it contains,
this also refers to other long lines

czajkaaga reviewed (commented):
Good job! :)

kristinapeneva committed:
After Method Added

kristinapeneva committed:
Created variable country and used it instead of hardcoding in the verifyShippingToCountryTest() method

kristinapeneva committed:
lines split for better readability

kristinapeneva committed:
Assertion messages added

kristinapeneva committed:
xPath shortened

kristinapeneva committed:
removed repetitions of 'Test'

kristinapeneva committed:
click() method replaced with clickButton method for better readability

kristinapeneva committed:
List<> - > List<String>

kristinapeneva committed:
10 -> Duration.ofSeconds(10) - better readability

kristinapeneva commented:
Hello,
I reworked my code according to your review.
You may check it once again :)

kristinapeneva committed:
- deleted clickButton method, not really useful;
- added deliverToButtonClick() method;
- added WebElement wait to each test to avoid repetition.

kristinapeneva commented:
Refactored. Ready for merging.

dominikbodzioch commented:
Just after switching to your branch Intellij immediately notifies me of a problem on line 28 (Webdrivermanager). 
From what I can see there is a proper import (line 3), but 'pom' does not provide the appropriate dependency - hence the signaled error.
After removing above mentioned lines everything is OK (Webdrivermanager is probably not necessary).
Lines 47, 89 and 92 may be split into 2 lines  but I probably shouldn't even mention that.
Good job Kristina!

kristinapeneva committed:
- WebDriverManager deleted as it's not necessary;
- split lines 46 and 89;

kristinapeneva commented:
Webdrivermanager deleted and i split the mentioned lines for readability reasons. Thanks, Dominik!

kristinapeneva committed:
- import for webdrivermanager deleted as well

czajkaaga commented on src/test/java/com/epam/qalab/DeliverToTest.java:
```diff
@@ -0,0 +1,110 @@
+
+public class DeliverToTest {
+    WebDriver driver;
+    String country = "Bulgaria";
```
this variable is only used in one test, so you can define it in this test (inside)

kristinapeneva replied:
done :)

dominikbodzioch replied:
Sometimes I leave these kinds of variables (like Bulgaria) "at the top of the class" (as Kristina originally did) when for some reason I need to change its values and play around with the code.
It is easier to find such a variable when it is not "hidden" in some method.
I don't know if I am breaking some clean code rules by doing this...
Marat could also have moved the "exec" variable to the "setUp()" method. But it was probably worth leaving it as a class variable (in my subjective opinion), because its value depends on the operating system (we can't run the test until it matches our OS, it should be easy to find).
In this case I simply don't know :)

kristinapeneva committed:
the variable country moved inside the test

juliaHrabovska commented on src/test/java/com/epam/qalab/DeliverToTest.java:
```diff
@@ -0,0 +1,111 @@
+import java.util.List;
+
+public class DeliverToTest {
+    WebDriver driver;
```
Can be private

juliaHrabovska commented on src/test/java/com/epam/qalab/DeliverToTest.java:
```diff
@@ -0,0 +1,111 @@
+        category.click();
+
+        WebElement keyboardItem = wait
+                .until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//*[contains(@class,'s-main-slot ')]/div[@data-component-type='s-search-result'][1]//h2/a")));
```
Not critical, but it is better not to too long line. I mean the size of the line should be 120, no more. So it is better to move some part to a new line
